x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/controlled_field.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiFormRow, EuiFormRowProps } from '@elastic/eui';
+import { useSelector } from 'react-redux';
+import {
+  UseFormReturn,
+  ControllerRenderProps,
+  ControllerFieldState,
+  useFormContext,
+} from 'react-hook-form';
+import { useKibanaSpace, useIsEditFlow } from '../hooks';
+import { selectServiceLocationsState } from '../../../state';
+import { FieldMeta } from './config';
+
+type Props = FieldMeta & {
+  component: React.ComponentType<any>;
+  field: ControllerRenderProps;
+  fieldState: ControllerFieldState;
+  formRowProps: Partial<EuiFormRowProps>;
+  error: React.ReactNode;
+  dependenciesValues: unknown[];
+  dependenciesFieldMeta: Record<string, ControllerFieldState>;
+};
+
+const setFieldValue = (key: string, setValue: UseFormReturn['setValue']) => (value: any) => {
+  setValue(key, value);
+};
+
+export const ControlledField = ({
+  component: Component,
+  props,
+  fieldKey,
+  useSetValue,
```

Comment:
Would it be better to disambiguate `useSetValue` to not be taken as a hook, i.e. to booleanize the prop name to `shouldUseSetValue`, `isUseSetValue` (or better)?